test/e2e/backup_restore_test.go

```diff
@@ -26,94 +27,102 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+type clusterSpec func(name, namespace string, clusterSize int) *v1.Infinispan
+
 func TestBackupRestore(t *testing.T) {
-	// Create a resource without passing any config
-	name := strcase.ToKebab(t.Name())
-	namespace := tconst.Namespace
-	clusterSize := 1
-	numEntries := 100
-
-	// 1. Create initial source cluster
-	sourceCluster := name + "-source"
-	infinispan := clusterSpec(sourceCluster, namespace, clusterSize)
-
-	testKube.Create(infinispan)
-	defer testKube.DeleteInfinispan(infinispan, tconst.SinglePodTimeout)
-	waitForPodsOrFail(infinispan, clusterSize)
-
-	// 2. Populate the cluster with some data to backup
-	protocol := getSchemaForRest(infinispan)
-	cluster := newCluster(cconsts.DefaultOperatorUser, infinispan.GetSecretName(), protocol, testKube.Kubernetes)
-	cacheName := "someCache"
-	populateCache(cacheName, sourceCluster+"-0", numEntries, cluster.Client)
-
-	// 3. Backup the cluster's content
-	backupSpec := &v2.Backup{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      name,
-			Namespace: namespace,
-		},
-		Spec: v2.BackupSpec{
-			Cluster: sourceCluster,
-		},
-	}
-	testKube.Create(backupSpec)
-	defer testKube.DeleteBackup(backupSpec)
+	t.Run(string(v1.ServiceTypeDataGrid), testBackupRestore(datagridService))
+	t.Run(string(v1.ServiceTypeCache), testBackupRestore(cacheService))
```

Comment:
Maybe not for all backup/restore tests (I plan to add more), but I think it would be good to have one covering the basics. I mainly wanted to ensure that the default cache did not cause issues on restore